### PR TITLE
Fix #3418: Workaround for specific XMLRPC methods replying with 403 errors

### LINKED ISSUE #3418: App acts weird if you're not an Editor or Admin on all of your blogs
A user reported that the app is unresponsive and sometime lags or exposes unexpected behavior if hte user logged in isn't an Admin or Editor on all sites.

> also the login shows an odd behaviour: After entering my credentials at the login page the app shows a black screen for some moments then again requests the credentials, and again, and again, ...
But if I cancel the login request I actually enter the app and my site/statistics/...

Here's the log that shows some 403 and 401 errors:

WordPress Android version: 4.6
Android device name: Samsung GT-I8160
```
01 - WordPress.onCreate
02 - trackLastActivity, activityId: Me
03 - App comes from background
04 - ConnectionChangeReceiver.setEnabled true
05 - Connection status changed, isConnected=true
06 - Deferred Initialisation
07 - Already registered for GCM
08 - Configuring Simperium
09 - Number of old stats deleted : 11
10 - Response document received from the server: <?xml version="1.0" encoding="UTF-8"?>
<methodResponse>
  <fault>
    <value>
      <struct>
        <member>
          <name>faultCode</name>
          <value><int>403</int></value>
        </member>
        <member>
          <name>faultString</name>
          <value><string>You are not allowed access to details about this site.</string></value>
        </member>
      </struct>
    </value>
  </fault>
</methodResponse>

11 - NETWORK_XMLRPC - You are not allowed access to details about this site. [Code: 403] - exception: You are not allowed access to details about this site. [Code: 403]
```
and later in the log
```
46 - Response document received from the server: <?xml version="1.0" encoding="UTF-8"?>
<methodResponse>
  <fault>
    <value>
      <struct>
        <member>
          <name>faultCode</name>
          <value><int>403</int></value>
        </member>
        <member>
          <name>faultString</name>
          <value><string>You are not allowed access to details about this site.</string></value>
        </member>
      </struct>
    </value>
  </fault>
</methodResponse>
```

### PR DESCRIPTION
**Pull request by maxme**
Fix #3418: Workaround for specific XMLRPC methods replying with 403 errors

### CODE REVIEW

File: WordPress/src/main/java/org/xmlrpc/android/XMLRPCClient.java [hunk 1]
```diff
@@ -534,6 +534,12 @@ private Object callXMLRPC(String method, Object[] params, File tempFile)
                 // Detect login issues and broadcast a message if the error is known
                 switch (e.getFaultCode()) {
                     case 403:
+                        // Ignore 403 error from certain methods known for replying with incorrect error code on
+                        // lacking permissions
+                        if ("wp.getPostFormats".equals(method) || "wp.getCommentStatusList".equals(method)
+                            || "wp.getPostStatusList".equals(method) || "wp.getPageStatusList".equals(method)) {
+                            break;
```
Comment (nbradbury) [inline WordPress/src/main/java/org/xmlrpc/android/XMLRPCClient.java]: What happens when the break statement is reached here? Does the API call simply fail silently, and the user never sees the desired info (post formats, etc.)?

Comment (maxme) [inline WordPress/src/main/java/org/xmlrpc/android/XMLRPCClient.java]: This call is done in the [RefreshBlogContentTask](https://github.com/wordpress-mobile/WordPress-Android/blob/bac388aac9284360069856505ee1c89c713211d6/WordPress/src/main/java/org/xmlrpc/android/ApiHelper.java#L229-L229), which is call when we switch to a blog or when we the app starts. The call fails silently, and the user can only select the default post formats when he wants to create a new post.

**Comment by maxme** [comment]
Minor, but last item lacks a bottom margin: https://cloudup.com/cb8T8EQbrW7

**Comment by nbradbury** [comment]
:shipit:

**Comment by maxme** [comment]
I'm seeing some Helpshift requests about this problem. I'll merge this fix in 4.8 (next week release).

**Pull request merged**